SPECS/zstd/zstd.spec

```diff
@@ -1,96 +1,323 @@
 
-%package        devel
-Summary:        Header and development files for zstd
-Requires:       %{name} = %{version}-%{release}
-Provides:       lib%{name}-devel = %{version}-%{release}
+%package -n lib%{name}
+Summary:        Zstd shared library
 
-%description    devel
-Header and development files for zstd compression
+# For backwards compatibility
+Provides:       zstd-libs = %{version}-%{release}
+Obsoletes:      zstd-libs <= 1.5.5-5
 
-%package        libs
-Summary:        Libraries for zstd
-Group:          System Environment/Libraries
-Provides:       lib%{name} = %{version}-%{release}
+%description -n lib%{name}
+Zstandard compression shared library.
 
-%description    libs
-This package contains minimal set of shared zstd libraries.
+%package -n lib%{name}-devel
+Summary:        Header files for Zstd library
+Requires:       lib%{name}%{?_isa} = %{version}-%{release}
 
-%package        doc
-Summary:        Documentation files for zstd
-Requires:       %{name} = %{version}-%{release}
+# For backwards compatibility
+Provides:       zstd-devel = %{version}-%{release}
+Obsoletes:      zstd-devel <= 1.5.5-5
 
-%description    doc
-Documentation files for zstd
+%package -n lib%{name}-static
+Summary:        Static variant of the Zstd library
+Requires:       lib%{name}-devel = %{version}-%{release}
+
+%description -n lib%{name}-devel
+Header files for Zstd library.
+
+%description -n lib%{name}-static
+Static variant of the Zstd library.
 
 %prep
-%autosetup
+%setup -q
+find -name .gitignore -delete
+%if %{with pzstd}
+%patch1 -p1
+%endif
 
 %build
-%make_build
+export CFLAGS="$RPM_OPT_FLAGS"
+export LDFLAGS="$RPM_LD_FLAGS"
+export PREFIX="%{_prefix}"
+export LIBDIR="%{_libdir}"
+%make_build -C lib lib-mt %{!?with_asm:ZSTD_NO_ASM=1}
+%make_build -C programs %{!?with_asm:ZSTD_NO_ASM=1}
+%if %{with pzstd}
+export CXXFLAGS="$RPM_OPT_FLAGS"
 %make_build -C contrib/pzstd %{!?with_asm:ZSTD_NO_ASM=1}
+%endif
+
+%check
+execstack lib/libzstd.so.1
+
+export CFLAGS="$RPM_OPT_FLAGS"
+export LDFLAGS="$RPM_LD_FLAGS"
+make -C tests test-zstd
+%if %{with pzstd} && %{with gtest}
+export CXXFLAGS="$RPM_OPT_FLAGS"
+make -C contrib/pzstd test
+%endif
 
 %install
-%make_install prefix=%{_prefix}
-find %{buildroot} -type f -name "*.a" -delete -print
+%make_install PREFIX=%{_prefix} LIBDIR=%{_libdir}
+%if %{with pzstd}
 install -D -m755 contrib/pzstd/pzstd %{buildroot}%{_bindir}/pzstd
+install -D -m644 programs/%{name}.1 %{buildroot}%{_mandir}/man1/p%{name}.1
+%endif
 
-%check
-%make_build check
+%files
+%doc CHANGELOG README.md
+%{_bindir}/%{name}
+%if %{with pzstd}
+%{_bindir}/p%{name}
+%{_mandir}/man1/p%{name}.1*
+%endif
+%{_bindir}/%{name}mt
+%{_bindir}/un%{name}
+%{_bindir}/%{name}cat
+%{_bindir}/%{name}grep
+%{_bindir}/%{name}less
+%{_mandir}/man1/%{name}.1*
+%{_mandir}/man1/un%{name}.1*
+%{_mandir}/man1/%{name}cat.1*
+%{_mandir}/man1/%{name}grep.1*
+%{_mandir}/man1/%{name}less.1*
+%license COPYING LICENSE
 
-%ldconfig_scriptlets
+%files -n lib%{name}
```

Comment:
Can we stick to the original `zstd-libs` name of the package? It causes unnecessary re-names in other spec but more importantly this may be breaking for people already using the original version. Alternatively, please use the `Provides` tag.